backend/api/Controllers/EchoController.cs

```diff
@@ -165,4 +169,47 @@ public async Task<ActionResult<EchoPlantInfo>> GetEchoPlantInfos()
             return new StatusCodeResult(StatusCodes.Status500InternalServerError);
         }
     }
+
+    /// <summary>
+    /// Get all plants associated with an active robot.
+    /// </summary>
+    /// <remarks>
+    /// <para> Retrieves the plants that have an active robot </para>
+    /// </remarks>
+    [HttpGet]
+    [Authorize(Roles = Role.User)]
+    [Route("active-plants")]
+    [ProducesResponseType(typeof(IList<EchoPlantInfo>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IList<EchoPlantInfo>>> GetActivePlants()
+    {
+        var plants = await _robotService.ReadAllActivePlants();
+
+        if (plants == null)
+        {
+            _logger.LogWarning("Could not retrieve robot plants information");
+            throw new RobotInformationNotAvailableException("Could not retrieve robot plants information");
+        }
+
+        try
+        {
+            var echoPlantInfos = await _echoService.GetEchoPlantInfos();
+
+            echoPlantInfos = echoPlantInfos.Where(p => plants.Contains(p.PlantCode)).ToList();
```

Comment:
I noticed if you write kaa instead of KAA the list does not show the Kårstø installation, so maybe add a ToLower here